tests/test_parser.py

```diff
@@ -20,10 +20,7 @@
 
 import linkcheck.HtmlParser.htmlsax
 import linkcheck.HtmlParser.htmllib
-try:
-    from cStringIO import StringIO
-except ImportError:
-    from io import StringIO
+from io import StringIO
```

Comment:
So I don't see how this improves things?

io.StringIO will be used on Python 3 regardless.

cStringIO.StringIO is more permissive than io.StringIO on Python 2: it accepts both bytes and unicode, while io.StringIO requires unicode.

Where io.StringIO is better is when you want to have early problem detection when you accidentally write something that's not unicode into your StringIO, instead of having it explode later when you call .getvalue().

Reply:
I'm afraid this was just the result of experimentation. All I can say is it fixed 9 test failures like the one above.

What I had seen in the Python 2 documentation for cStringIO.StringIO was:
"Unlike the StringIO module, this module is not able to accept Unicode strings that cannot be encoded as plain ASCII strings."